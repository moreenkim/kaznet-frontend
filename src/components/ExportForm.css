.export-form select.form-control:not([size]):not([multiple]) {
  height: calc(2.25rem + 2px);
  width: 395px;
  position: relative;
}
