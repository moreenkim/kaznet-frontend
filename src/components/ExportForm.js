import React, { Component } from "react";
import { Col, Input, FormGroup, Form, Button } from "reactstrap";
import { Formik, Field } from "formik";
import moment from "moment";
import * as constants from "../constants";
import "./ExportForm.css";

export default class ExportForm extends Component {
  render() {
    return (
      <Formik
        initialValues={{
          start: moment().format("YYYY-MM-DD"),
          end: moment().format("YYYY-MM-DD"),
          status: constants.SUBMISSION_APPROVED
        }}
        onSubmit={(values, { setSubmitting, setErrors, setStatus }) => {
          this.props.onFormSubmit(values.start, values.end, values.status);
        }}
        render={({ values, handleChange, handleBlur, handleSubmit }) => (
          <div>
            <Form onSubmit={handleSubmit}>
              <FormGroup row>
                <Col md={{ size: 1 }}>
                  <p className="text-center align-middle">Status:</p>
                </Col>
                <Col md={{ size: 10 }}>
                  <Field
                    name="status"
                    component="select"
                    className="form-control ml-4"
                  >
                    <option value={constants.SUBMISSION_APPROVED}>
                      Approved
                    </option>
                    <option value={constants.SUBMISSION_REJECTED}>
                      Rejected
                    </option>
                    <option value={constants.SUBMISSION_PENDING}>
                      Pending
                    </option>
                  </Field>
                </Col>
              </FormGroup>
              <FormGroup row>
                <p className="text-center align-middle ml-4">From:</p>
                <Col md={{ size: 5 }}>
                  <Input
                    name="start"
                    type="date"
                    bsSize="md"
                    placeholder="Start Date"
                    aria-label="start"
                    className={`time-picker`}
                    onChange={handleChange}
                    onBlur={handleBlur}
                    value={values.start}
                  />
                </Col>
                <p className="text-center align-middle">to</p>
                <Col md="5">
                  <Input
                    name="end"
                    type="date"
                    bsSize="md"
                    placeholder="End Date"
                    aria-label="end"
                    className={`time-picker`}
                    onChange={handleChange}
                    onBlur={handleBlur}
                    value={values.end}
                  />
                </Col>
              </FormGroup>
              <FormGroup row>
                <Col md={{ size: 5, offset: 1 }}>
                  <Button
                    className="btn btn-secondary btn-block"
                    color="secondary"
                    onClick={this.props.downloadModalHandler}
                    aria-label="Cancel"
                  >
                    Cancel
                  </Button>
                </Col>
                <Col md={{ size: 5 }}>
                  <Button
                    type="submit"
                    className="btn btn-primary btn-block"
                    color="secondary"
                    onClick={this.props.downloadModalHandler}
                    aria-label="Export"
                  >
                    Export
                  </Button>
                </Col>
              </FormGroup>
            </Form>
          </div>
        )}
      />
    );
  }
}
