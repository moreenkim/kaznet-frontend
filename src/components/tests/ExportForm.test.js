// test ExportForm
import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import ExportForm from "../ExportForm";

describe("components/ExportForm", () => {
  it("renders without crashing", () => {
    shallow(
      <ExportForm
        onFormSubmit={function() {}}
        handleDateChanges={function() {}}
        downloadModalHandler={function() {}}
      />
    );
  });
  it("renders an ExportForm", () => {
    const wrapper = mount(
      <ExportForm
        onFormSubmit={function() {}}
        handleDateChanges={function() {}}
        downloadModalHandler={function() {}}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
