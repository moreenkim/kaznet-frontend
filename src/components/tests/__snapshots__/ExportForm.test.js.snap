// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ExportForm renders an ExportForm 1`] = `
<ExportForm
  downloadModalHandler={[Function]}
  onFormSubmit={[Function]}
>
  <Formik
    enableReinitialize={false}
    initialValues={
      Object {
        "end": "1986-01-02",
        "start": "1986-01-02",
        "status": "a",
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    render={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
  >
    <div>
      <Form
        className="export-form"
        onSubmit={[Function]}
        tag="form"
      >
        <form
          className="export-form"
          onSubmit={[Function]}
        >
          <FormGroup
            row={true}
            tag="div"
          >
            <div
              className="position-relative row form-group"
            >
              <Col
                md={
                  Object {
                    "size": 1,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-md-1"
                >
                  <p
                    className="text-center align-middle"
                  >
                    Status:
                  </p>
                </div>
              </Col>
              <Col
                md={
                  Object {
                    "size": 10,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-md-10"
                >
                  <Field
                    className="form-control ml-4"
                    component="select"
                    name="status"
                  >
                    <select
                      className="form-control ml-4"
                      name="status"
                      onBlur={[Function]}
                      onChange={[Function]}
                      value="a"
                    >
                      <option
                        value="a"
                      >
                        Approved
                      </option>
                      <option
                        value="b"
                      >
                        Rejected
                      </option>
                      <option
                        value="d"
                      >
                        Pending
                      </option>
                    </select>
                  </Field>
                </div>
              </Col>
            </div>
          </FormGroup>
          <FormGroup
            row={true}
            tag="div"
          >
            <div
              className="position-relative row form-group"
            >
              <p
                className="text-center align-middle ml-4"
              >
                From:
              </p>
              <Col
                md={
                  Object {
                    "size": 5,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-md-5"
                >
                  <Input
                    aria-label="start"
                    bsSize="md"
                    className="time-picker"
                    name="start"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Start Date"
                    type="date"
                    value="1986-01-02"
                  >
                    <input
                      aria-label="start"
                      className="time-picker form-control-md form-control"
                      name="start"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Start Date"
                      type="date"
                      value="1986-01-02"
                    />
                  </Input>
                </div>
              </Col>
              <p
                className="text-center align-middle"
              >
                to
              </p>
              <Col
                md="5"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-md-5"
                >
                  <Input
                    aria-label="end"
                    bsSize="md"
                    className="time-picker"
                    name="end"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="End Date"
                    type="date"
                    value="1986-01-02"
                  >
                    <input
                      aria-label="end"
                      className="time-picker form-control-md form-control"
                      name="end"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="End Date"
                      type="date"
                      value="1986-01-02"
                    />
                  </Input>
                </div>
              </Col>
            </div>
          </FormGroup>
          <FormGroup
            row={true}
            tag="div"
          >
            <div
              className="position-relative row form-group"
            >
              <Col
                md={
                  Object {
                    "offset": 1,
                    "size": 5,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-md-5 offset-md-1"
                >
                  <Button
                    aria-label="Cancel"
                    className="btn btn-secondary btn-block"
                    color="secondary"
                    onClick={[Function]}
                    tag="button"
                  >
                    <button
                      aria-label="Cancel"
                      className="btn btn-secondary btn-block btn btn-secondary"
                      onClick={[Function]}
                      type="button"
                    >
                      Cancel
                    </button>
                  </Button>
                </div>
              </Col>
              <Col
                md={
                  Object {
                    "size": 5,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-md-5"
                >
                  <Button
                    aria-label="Export"
                    className="btn btn-primary btn-block"
                    color="secondary"
                    onClick={[Function]}
                    tag="button"
                    type="submit"
                  >
                    <button
                      aria-label="Export"
                      className="btn btn-primary btn-block btn btn-secondary"
                      onClick={[Function]}
                      type="submit"
                    >
                      Export
                    </button>
                  </Button>
                </div>
              </Col>
            </div>
          </FormGroup>
        </form>
      </Form>
    </div>
  </Formik>
</ExportForm>
`;
